mlflow/tracking/fluent.py

```diff
@@ -686,20 +715,34 @@ def log_metric(key: str, value: float, step: Optional[int] = None) -> None:
                   All backend stores will support values up to length 5000, but some
                   may support larger values.
     :param step: Metric step (int). Defaults to zero if unspecified.
+    :param synchronous: [Experimental] Defaults to True.
+                        Indicates if the metric would be logged in synchronous fashion or not.
+                    When it is True this call would be blocking call and offers immediate
+                        consistency of the metric upon returning.
+                    When this value is set to False, metric would be logged in async
+                        fashion. Backing provider gurantees that metrics are accepted
+                        into system but would persist them with some time delay -
+                        'near-real time'/'eventual' consistency.
+    :return: Returns RunOperations object. This object, when not None,
+                can be awaited to know if given metric was processed or not.
 
-    .. testcode:: python
         :caption: Example
 
         import mlflow
 
         with mlflow.start_run():
             mlflow.log_metric("mse", 2500.00)
+            mlflow.log_metric("mse", 2500.00, synchronous=False)
```

Comment:
Let's make a new example, we may not want to suggest mixing async logging with sync logging inside the same mlflow run.